impl/src/main/java/org/ehcache/impl/internal/store/disk/OffHeapDiskStore.java

```diff
@@ -466,6 +463,7 @@ public void initAuthoritativeTier(AuthoritativeTier<?, ?> resource) {
    * This is kind of a hack, but it's safe to use this if the regular portability
    * is stateless.
    */
+  @SuppressWarnings("unchecked")
```

Comment:
The problem with the annotation is that you cannot always be as
fine-grained at wanted. And it can be `Util.uncheckedCast(Object object,
String why)` :-)

On 12 September 2016 at 11:37, Louis Jacomet <notifications@github.com>
wrote:

> In impl/src/main/java/org/ehcache/impl/internal/store/
> disk/OffHeapDiskStore.java
> <https://github.com/ehcache/ehcache3/pull/1431#discussion_r78396707>:
>
> > @@ -466,6 +463,7 @@ public void initAuthoritativeTier(AuthoritativeTier<?, ?> resource) {
> >     * This is kind of a hack, but it's safe to use this if the regular portability
> >     * is stateless.
> >     */
> > +  @SuppressWarnings("unchecked")
>
> I prefer having the annotation exist where it is really needed. I hate the
> fact that the annotation does not have an attribute to explain why you are
> using it btw.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ehcache/ehcache3/pull/1431/files/df5125b54078e9884ffbacc956259b900668b93a#r78396707>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABKRM8uXQgcttklis2BetL3zKc4u8GK0ks5qpXG3gaJpZM4J6bKp>
> .
>